keeper/vault.go

```diff
@@ -197,12 +201,13 @@ func (k *Keeper) SwapIn(ctx sdk.Context, vaultAddr, recipient sdk.AccAddress, as
 // SwapOut handles the process of redeeming vault shares in exchange for underlying assets.
 // It performs the following steps:
 //  1. Retrieves the vault configuration for the given vault address.
-//  2. Validates that the provided share denomination matches the vault's configured share denom.
-//  3. Calculates the amount of underlying assets to return based on the share amount.
-//  4. Transfers the shares from the owner to the vault's marker account.
-//  5. Burns the received shares from the vault account.
-//  6. Sends the equivalent amount of underlying assets from the marker to the owner.
-//  7. Emits a SwapOut event with metadata for indexing and audit.
+//  2. Reconciles any accrued interest from the vault to the marker module if due.
+//  3. Validates that the provided share denomination matches the vault's configured share denom.
+//  4. Calculates the amount of underlying assets to return based on the share amount.
+//  5. Transfers the shares from the owner to the vault's marker account.
+//  6. Burns the received shares from the vault account.
+//  7. Sends the equivalent amount of underlying assets from the marker to the owner.
+//  8. Emits a SwapOut event with metadata for indexing and audit.
 //
 // Returns the burned share amount on success, or an error if any step fails.
 func (k *Keeper) SwapOut(ctx sdk.Context, vaultAddr, owner sdk.AccAddress, shares sdk.Coin) (*sdk.Coin, error) {
```

Comment:
It would be good to see if we can test to ensure that if this action fails, none of the reconcile changes are persisted either (the movement of funds rolls back, the start of period doesn't change, etc.

Reply:
That is what I expect… but it would be a good thing to verify. I believe there is a snapshot that wraps the state store (for a tx) so everything written to it is abandoned on failure. We should be extra careful of any changes that might error in our begin/end block functions as they don’t have the same error and rollback protections if I remember correctly.

Reply:
Yes, I was thinking about that too with the begin/end blocker.